libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -25,10 +28,16 @@ const AP_Param::GroupInfo AP_BattMonitor::var_info[] = {
 // Note that the Vector/Matrix constructors already implicitly zero
 // their values.
 //
-AP_BattMonitor::AP_BattMonitor(void) :
+AP_BattMonitor::AP_BattMonitor(uint32_t log_battery_bit) :
+    _log_battery_bit(log_battery_bit),
     _num_instances(0)
 {
     AP_Param::setup_object_defaults(this, var_info);
+
+    if (_singleton != nullptr) {
+        AP_HAL::panic("AP_BattMonitor must be singleton");
```

Comment:
I think we have several ways of doing this in our code: some just always panic like here, others only panic in SITL. I prefer the later, but I'm not sure if people care - tagging for the dev call.